tree/ntuple/v7/inc/ROOT/RPageStorage.hxx

```diff
@@ -175,6 +206,14 @@ public:
    EPageStorageType GetType() final { return EPageStorageType::kSource; }
    const RNTupleDescriptor &GetDescriptor() const { return fDescriptor; }
    ColumnHandle_t AddColumn(DescriptorId_t fieldId, const RColumn &column) final;
+   void DropColumn(ColumnHandle_t columnHandle) final;
+
+   /// During the life time of a column reader, we may have different periods with different
+   /// sets of active columns.  For instance, RNTupleReader::Show() will temporarily connect all columns.
+   void StashColumns();
+   /// Swap in a previously stashed set of columns and replace the current set of active columns
+   void PopColumns();
+   RColumnGuard GetColumnGuard() { return RColumnGuard(this); }
```

Comment:
I am not overly fond of this pattern.   It is used to handle the CollectionProxy in multiple place and have since need to work around it to improve thread safety/performance.   Is there a way to make it an active/explicit request (for example pass a set of columns rather than having a 'global' state)?

Reply:
That's a tricky one. The `StashColumns` and `PopColumns` are certainly not meant to be taken in MT context.  I see them as a larger reconfiguration that may happen between longer (multi-threaded) runs with a stable set of active columns.

This is currently used for `RNTupleReader::Show()`: in interactive use, the user might look into a few entries and display everything of those.  Afterwards, the user might run an analysis covering only a few columns.  This occurred to me while writing the ntpl005 tutorial.

Reply:
I might have found a better solution: the printout of data (`RNTupleReader::Show()`) is now treated as an out-of-band access by using a second on-demand reader, which is a clone of the original one but with all the columns connected. The code for stashing of active columns is removed.

Reply:
I can imagine there are use cases for both full entry printing and printing the currently active model.  I added the `kJSONFull` flag that can be provided to indicate that the entire entry should be shown.  By default, `Show()` will now only print the fields defined in the reader's model.

Reply:
Fair enough.  Do we need the same for Scan?

Reply:
Just for the record, as discussed in the I/O meeting, `Scan()` is missing entirely for the time being.